modules/renter/download.go

```diff
@@ -491,11 +491,11 @@ func (r *Renter) DownloadHistory() []modules.DownloadInfo {
 	defer r.downloadHistoryMu.Unlock()
 
 	downloads := make([]modules.DownloadInfo, len(r.downloadHistory))
-	for i := range r.downloadHistory {
+	i := 0
+	for _, d := range r.downloadHistory {
```

Comment:
That won't work I think. `d` starts from the first element while you use `i` to traverse the list from the back. That means you don`t hole the correct lock. You wouldn't notice if the length is 1 but for a longer list that might cause problems.